trie/iterator.go

```diff
@@ -144,7 +144,8 @@ type nodeIterator struct {
 	path  []byte               // Path to the current node
 	err   error                // Failure set in case of an internal error in the iterator
 
-	resolver NodeResolver // optional node resolver for avoiding disk hits
+	resolver NodeResolver         // optional node resolver for avoiding disk hits
+	pool     []*nodeIteratorState // local pool for iteratorstates
```

Comment:
Well, it's rather that I didn't think a global `nodeIteratorPool` was "generically useful". We only use node iterators here and there. So I figured that we can take the overhead of 'ramping up' a fresh pool in those instances, rather than having yet another global pool. 

Aside from that, a local non-synched pool is slightly faster since it doesn't need syncing.